Solutions/LargestPrimeFactor.cs

```diff
@@ -0,0 +1,36 @@
+namespace Solutions;
+
+public class LargestPrimeFactor : ProblemSolverBase
+{
+  public LargestPrimeFactor()
+  {
+    ProblemID = 3;
+  }
+  public override void Solve()
+  {
+    Answer = GetLargestPrimeFactor(600851475143).ToString();
+  }
+
+  long GetLargestPrimeFactor(long number)
+  {
+    long largestPrime = 0;
+    long divisor = 2;
+
+    while (number != 1)
+    {
+      if (number % divisor == 0)
+      {
+        number = number / divisor;
+      }
+      else
+      {
+        divisor++;
```

Comment:
I think instead of going with this approach of just brute forcing every number there is, why not just try to generate the next prime instead? Right now this is O(n), you should be able to do a lot better. 

So you'd have two utility functions - one that generates the prime and this one which would use the prime generation function. 

A hint on the prime generation function, it doesn't necessarily have to be a function. I think it should be a class that has multiple utilities and memoizes the already calculated primes by index. 

It should: 
- generate the next prime on demand (so it is not a list but a generator instead)
- keep the previous primes in memory (this is already done by the best algorithms for speed reasons but your should have it public where you can access by some method, like primes.getByIndex(10) which would return the 10th prime. 
- would be nice if it had something like .isPrime which would check the size of the number and if it is less than the biggest prime you keep it would check if it is in the calculated prime list, if it is not it would generate prime numbers that goes one above to the given number and see if it is _now_ in the generated numbers
- the above naturally calls for a method .generateUpTo(long) which generates primes and adds them to the internal list 
- also it would be cool if it had a cursor that you can control like below (it is not c# syntax, it is pseudo code because idk c#)
```
 primes = PrimeGenerator()
 primes.next() # 2 
 primes.next() # 3 
 primes.next() # 5
 primes.next() # 7
 primes.seek(1) # cursor is now at index 1, prime at index 1 is 3, next one should be 5
 primes.next() # 5
 
 ```
 
 Feel free to add anything that makes sense to you that you think would be helpful in the future or would be cool